Assets/Scripts/Stats/Health.cs

```diff
@@ -1,23 +1,31 @@
 using Assets.Scripts.Stats;
-using System;
 using System.Collections;
-using System.Collections.Generic;
+using Unity.Collections;
 using UnityEngine;
 using UnityEngine.Events;
 
-public class Health : Stat
+public class Health : RegeneratingStats
 {
-    [SerializeField] bool _isInvincible;
+    [SerializeField] float _regenRateInSeconds;
+    [SerializeField] Armor _armor;
+    [SerializeField, ReadOnly] bool _isInvincible;
+    [SerializeField, ReadOnly] bool _isDead;
```

Comment:
Would be more efficient to use a property 
`IsDead => health <= 0`
Otherwise, if your goal was to allow me to kill an entity from the inspector, it shouldn't be read-only.